### PR TITLE
mobile: install Android application context at start up

### PR DESCRIPTION
**Pull request by danzh2010**
Commit Message: call `ContextUtils.initApplicationContext()` in AndroidEngineImpl to initialize the static member in `ContextUtil` class which is used by the original cronet libraries like `AndroidNetworkLibrary`.

This fix a null pointer crash while using Android platform cert verification API.

Risk Level: low
Testing: modified `FakeX509Util` to check this context to prevent regression
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: mobile

### CODE REVIEW

**Review by abeyad — DISMISSED** [review]

**Comment by abeyad** [comment]
@danzh2010 the failing CI check looks like a legitimate test failure: https://github.com/envoyproxy/envoy/actions/runs/4856952680/jobs/8656995376?pr=27093

File: mobile/examples/java/hello_world/MainActivity.java [hunk 1]
```diff
@@ -61,6 +63,7 @@ protected void onCreate(Bundle savedInstanceState) {
                    Log.d("MainActivity", "Envoy async internal setup completed");
                    return null;
                  })
+                 // .enablePlatformCertificatesValidation(true)
```
Comment (abeyad) [inline mobile/examples/java/hello_world/MainActivity.java:66]: was this commented out on purpose?

Comment (danzh2010) [reply mobile/examples/java/hello_world/MainActivity.java:66]: Actually https requests interleaving with http request doesn't work neither with nor without platform cert validator. But running them alone worked. I don't know the reason yet. I will split out this change and let the rest get in first.

**Review by RyanTheOptimist — APPROVED** [review]

**Pull request merged**